# Changes related to golang-github-aws-aws-sdk-go latest upload
# Patch by Filippo Giunchedi <filippo@debian.org>

--- a/discovery/ec2/ec2.go
+++ b/discovery/ec2/ec2.go
@@ -21,7 +21,7 @@
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/defaults"
+	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/prometheus/common/log"
 	"github.com/prometheus/common/model"
 	"golang.org/x/net/context"
@@ -57,7 +57,7 @@
 func NewEC2Discovery(conf *config.EC2SDConfig) *EC2Discovery {
 	creds := credentials.NewStaticCredentials(conf.AccessKey, conf.SecretKey, "")
 	if conf.AccessKey == "" && conf.SecretKey == "" {
-		creds = defaults.DefaultChainCredentials
+		creds = credentials.NewEnvCredentials()
 	}
 	return &EC2Discovery{
 		aws: &aws.Config{
@@ -100,7 +100,7 @@
 }
 
 func (ed *EC2Discovery) refresh() (*config.TargetGroup, error) {
-	ec2s := ec2.New(ed.aws)
+	ec2s := ec2.New(session.New(), ed.aws)
 	tg := &config.TargetGroup{
 		Source: *ed.aws.Region,
 	}
